### ISSUE TITLE
Unknown column 'submitActionEntryId' in 'field list' during migration

### ISSUE THREAD

**Issue by iainsaxon** [verbb/formie#858]
**Description**
I encountered an error while trying to update Formie from v1.4.24 to v1.5.15 during the database migrations:

```
*** applying m211211_000000_stencil_uids
Exception: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'submitActionEntryId' in 'field list'
The SQL being executed was: SELECT `id`, `name`, `handle`, `data`, `templateId`, `submitActionEntryId`, `defaultStatusId`, `dateDeleted`, `uid`
FROM `craft_formie_stencils`
WHERE `dateDeleted` IS NULL
ORDER BY `name` (/usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/db/Schema.php:676)
#0 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException(Object(PDOException), 'SELECT `id`, `n...')
#1 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute('SELECT `id`, `n...')
#2 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/db/Command.php(410): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/db/Query.php(152): yii\db\Query->all(NULL)
#5 /usr/local/var/www/victorchang.localhost/vendor/verbb/formie/src/services/Stencils.php(62): craft\db\Query->all()
#6 /usr/local/var/www/victorchang.localhost/vendor/verbb/formie/src/migrations/m211211_000000_stencil_uids.php(29): verbb\formie\services\Stencils->getAllStencils()
#7 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/db/Migration.php(52): verbb\formie\migrations\m211211_000000_stencil_uids->safeUp()
#8 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#9 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/controllers/MigrateController.php(386): yii\console\controllers\BaseMigrateController->migrateUp('m211211_000000_...')
#10 [internal function]: craft\console\controllers\MigrateController->actionAll()
#11 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#12 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#13 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('all', Array)
#14 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('all', Array)
#15 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#16 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/Application.php(89): yii\console\Application->runAction('migrate/all', Array)
#17 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Controller.php(212): craft\console\Application->runAction('migrate/all', Array)
#18 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/Controller.php(212): yii\base\Controller->run('migrate/all', Array)
#19 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/controllers/UpController.php(58): craft\console\Controller->run('migrate/all', Array)
#20 [internal function]: craft\console\controllers\UpController->actionIndex()
#21 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#22 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#23 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('', Array)
#24 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/Controller.php(221): yii\console\Controller->runAction('', Array)
#25 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Module.php(552): craft\console\Controller->runAction('', Array)
#26 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('up', Array)
#27 /usr/local/var/www/victorchang.localhost/vendor/craftcms/cms/src/console/Application.php(89): yii\console\Application->runAction('up', Array)
#28 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('up', Array)
#29 /usr/local/var/www/victorchang.localhost/vendor/yiisoft/yii2/base/Application.php(384): yii\console\Application->handleRequest(Object(craft\console\Request))
#30 /usr/local/var/www/victorchang.localhost/craft(8): yii\base\Application->run()
#31 {main}
*** failed to apply m211211_000000_stencil_uids (time: 0.378s)
```

I suspect the issue is the PHP model class has an attribute for `submitActionEntryId` that is expected to be in the database however this column is not added until the migration `m220420_000000_stencil_add_entryid` in the latest release (v1.5.15). When the ORM gets the data using the expected model attributes it adds the non-existent column to the SQL Select statement and fails.

To work around this I needed to first upgrade to Formie v1.5.2 first.

**Steps to reproduce**
1. Use an database backup and Formie version earlier than v1.5.2
2. Update to Formie v1.5.15 via `composer`
3. Run `craft up` to perform the migrations
4. Migration upgrade process will fail at m211211_000000_stencil_uids

**Additional info**
- Plugin version: v1.5.15
- Craft version: 3.7.27.2
- Multi-site: No

**Comment by engram-design** [comment]
Ah, good point. I missed the reference to fetching all stencils in a [past migration](https://github.com/verbb/formie/blob/craft-3/src/migrations/m211211_000000_stencil_uids.php), which I shouldn't have done that way in the first place as I've run into this situation before.

Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.15",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.16](https://github.com/verbb/formie/releases/tag/1.5.16)
